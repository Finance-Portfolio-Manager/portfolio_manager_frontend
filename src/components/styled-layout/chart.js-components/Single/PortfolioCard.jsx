import { chart } from "./PortfolioCard";
import { Doughnut } from "react-chartjs-2";
import React, { useState, useEffect } from "react";
import { Card } from "antd";
export const PortfolioCard = (trans, labels) => {
  const [state, setstate] = useState({ ...trans.transaction });
  const [label, setLabel] = useState({...labels});


 /*  const plugins = {
    plugins: [{
        beforeDraw: beforeDrawFn()
    }]};
  const beforeDrawFn = (chart) => {
    let width = chart.chart.width,
        height = chart.chart.height,
        ctx = chart.chart.ctx;

    ctx.restore();
    let fontSize = (height / 114).toFixed(2);
    ctx.font = fontSize + "em sans-serif";
    ctx.textBaseline = "middle";

    let text = "75%",
        textX = Math.round((width - ctx.measureText(text).width) / 2),
        textY = height / 2;

    ctx.fillText(text, textX, textY);
    ctx.save();
  } */



  //chart js set up
  const options =  {
    cutout: 99,
    responsive: true,
    plugins: {
      title: {
        display: true,
        text: state?.ticker,//Ticker
        align: 'center',
        position: 'bottom',
        
        padding: {
          // top: 10,
          // bottom: 30
      }
    },
        legend: {
            display: false,
            labels: {
                color: 'rgb(150, 251, 0)'
            }
        }
    }
  };

  //shareAmount: 100.0,
  // sharePrice: 200.0,
  // const labels = ["Share Amount", "Share Price"];//THIS
  const data = {
    labels: labels,
    datasets: [
      {
        label: "My First dataset",
        backgroundColor: [
          'rgb(255, 99, 132)',
          'rgb(54, 162, 235)',
          'rgb(255, 205, 86)'
        ],//here
        borderColor: [
          'rgb(255, 99, 132)',
          'rgb(54, 162, 235)',
          'rgb(255, 205, 86)'
        ], //here
        data: [state.shareAmount, state.sharePrice],//THIS
      },
    ],
    
    
  };

  const config = {
    type: 'doughnut',
    data: data,
    options: options
  };

  return (
    <>
      {console.log(state)}
      <div className="site-card-border-less-wrapper">
        <Card bordered={false} style={{ width: 300 }}>
          <div>
            <Doughnut data={data} options={options} 
            
            />
          </div>
        </Card>
      </div>
      {/* <chart></chart> */}
    </>
  );
};
