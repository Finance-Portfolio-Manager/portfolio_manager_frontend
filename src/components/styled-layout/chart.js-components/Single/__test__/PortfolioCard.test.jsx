import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PortfolioCard from './../PortfolioCard';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = shallow(<PortfolioCard></PortfolioCard>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<PortfolioCard></PortfolioCard>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("header renders", ()=>{
    expect(wrapper.find("h1").text()).toContain("@Username");
})