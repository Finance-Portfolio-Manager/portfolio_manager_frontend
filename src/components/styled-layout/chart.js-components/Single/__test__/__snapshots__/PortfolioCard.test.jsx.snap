// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="site-card-border-less-wrapper"
>
  <div
    className="ant-card card-body"
    style={
      Object {
        "width": 300,
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <h1
        className="card-header"
      >
        @Username
      </h1>
      <div
        className="chart-container"
      >
        <canvas
          className="zoom"
          data-testid="canvas"
          height={150}
          onClick={[Function]}
          role="img"
          width={300}
        />
      </div>
      <h2
        className="card-header"
      >
        $118.99
      </h2>
    </div>
  </div>
</div>
`;
